File: 03.match.ts
function describeNumber(n: number): string {
  switch(n) {
    case 1:
      return "One";
    case 2:
    case 3:
    case 5:
    case 7:
    case 11:
      return "Prime number";
    default:
      if (n >= 13 && n <= 19) {
        return "Teen number";
    }
    return "Something else";
  }
}

console.log(describeNumber(7));
console.log(describeNumber(15));
console.log(describeNumber(42));

File: 03.match.py
def describe_number(n):
    match n:
        case 1:
            return "One"
        case 2 | 3 | 5 | 7 | 11:
            return "Prime number"
        case _ if 13 <= n <= 19:
            return "Teen number"
        case _:
            return "Something else"

print(describe_number(7))
print(describe_number(15))
print(describe_number(42))
